auth-web/src/components/auth/staff/DissolutionSchedule.vue

```diff
@@ -0,0 +1,296 @@
+                      </v-list-item-subtitle>
+                    </v-list-item>
+                  </v-list>
+                </v-menu>
+              </span>
+            </v-col>
+          </template>
+        </v-row>
+
+        <!-- Cancel/Save buttons -->
+        <template v-if="isEdit">
+          <v-divider class="mb-8 mt-8" />
+          <v-row no-gutters>
+            <v-col
+              cols="12"
+              class="text-right"
+            >
+              <v-btn
+                color="primary"
+                large
+                outlined
+                class="mr-3"
+                @click="triggerEditOnOff()"
+              >
+                Cancel
+              </v-btn>
+              <v-btn
+                color="primary"
+                large
+                @click="saveBtnClicked()"
+              >
+                Save
+              </v-btn>
+            </v-col>
+          </v-row>
+        </template>
+      </article>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Vue } from 'vue-property-decorator'
+import { mapActions } from 'pinia'
+import { useStaffStore } from '@/stores/staff'
+
+@Component({
+  methods: {
+    ...mapActions(useStaffStore,
+      ['getDissolutionBatchSize',
+        'isDissolutionBatchOnHold',
+        'updateDissolutionBatchSize',
+        'updateDissolutionBatchOnHold'])
+  }
+})
+export default class DissolutionSchedule extends Vue {
+  $refs: {
+    numberOfBusinessesRef: any
+  }
+
+  // Local properties
+  menu = false
+  numberOfBusinesses = 0
+  isEdit = false
+  isOnHold = false
+
+  readonly getDissolutionBatchSize!: () => number
+  readonly isDissolutionBatchOnHold!: () => boolean
+  readonly updateDissolutionBatchSize!: (x: number) => void
+  readonly updateDissolutionBatchOnHold!: (x: boolean) => void
+
+  /**
+   * Set local properties to values from the store.
+   * TODO: Update once BE work is done.
+   */
+  mounted (): void {
+    this.numberOfBusinesses = this.getDissolutionBatchSize()
+    this.isOnHold = this.isDissolutionBatchOnHold()
+  }
+
+  /**
+   * Update whether the dissolution batch is paused or running.
+   * Emit the status to the parent to know whether the "Paused" badge is going to be shown.
+   * TODO: Modify/Update this once the BE is done. */
+  actionBtnClicked (): void {
+    this.isOnHold = !this.isDissolutionBatchOnHold()
+    this.onHoldUpdate(this.isOnHold)
+    this.updateDissolutionBatchOnHold(!this.isDissolutionBatchOnHold())
+  }
+
+  /**
+   * Save button is clicked. Update the dissolution batch size.
+   * Only save if the inputted number is valid.
+   * TODO: Implement logic (job) once the BE is done.
+   */
+  saveBtnClicked (): void {
+    if (this.$refs.numberOfBusinessesRef.validate()) {
+      this.updateDissolutionBatchSize(this.numberOfBusinesses)
+      this.triggerEditOnOff()
+    }
+  }
+
+  /** Edit or Cancel button is clicked. */
+  triggerEditOnOff (): void {
```

Comment:
The seems a little suss because it's used in so many ways. Passing in the new isEdit value might be easier to understand and test. Did you test all possible flows?

Reply:
Yes. I've tested thoroughly all the flows.

I thought the comment basically explains it. It'll be triggered when the Edit, Cancel, or Save (successful) buttons are clicked to show the extra fields.

On that note, I'll update the comment to include the Save button as well. Thanks for bringing it up!